file.c

```diff
@@ -533,7 +533,11 @@ filewclose(File *f)
     if (!f) return;
     if (!f->iswopen) return;
     if (f->free) {
-        (void)ftruncate(f->fd, f->w->filesize - f->free);
+        errno = 0;
+        int r = ftruncate(f->fd, f->w->filesize - f->free);
+        if (r == -1) {
+            twarn("ftruncate failed, errno=%d", errno);
```

Comment:
Is f->path available here? Could we specify a file name here as well?

Reply:
btw, 
```c
if (ftruncate(f->fd, f->w->filesize - f->free) != 0) {
    errno = 0;
    twarn("ftruncate failed, errno=%d", errno);
}
```
a bit shorter and w/o extra variable :)

Reply:
You are right about messages, they should be improved but it should be done more systematically and as separate effort. So for now I will skip adding f->path.
Thanks.

Reply:
Note that twarn mostly just calls the stdlib function warn, which has errno/strerror built in.

```
The err(), errc(), verr(), verrc(), warn(), warnc(), vwarn(), and
vwarnc() functions append an error message obtained from strerror(3)
based on a supplied error code value or the global variable errno, pre-
ceded by another colon and space unless the fmt argument is NULL.
```